.github/workflows/stage-2-test.yaml

```diff
@@ -95,6 +97,43 @@ jobs:
           DATABASE_USER: postgres
           DATABASE_HOST: localhost
 
+  notifications-integration-test:
+    needs: test
+    if: needs.test.outputs.notifications-changed
+    name: 'Notifications Integration Tests'
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    services:
+      postgres:
+        image: postgres:17.4-alpine3.21
+        ports:
+          - 5432:5432
+        env:
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - name: 'Checkout code'
+        uses: actions/checkout@v5
+
+      - name: Install poetry
+        run: pipx install poetry
+
+      - name: Set up Python
+        uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065
+        with:
+          python-version: '${{inputs.python_version}}'
+          cache: poetry
+          cache-dependency-path: ./poetry.lock
+
+      - name: Install dependencies
+        run: make dependencies
+
       - name: 'Run notifications integration test suite'
         if: steps.filter.outputs.notifications == 'true'
```

Comment:
Does this `if` statement needs removing? we've already got the `if` on the new job, and the `filter` id no longer exists?